localstack/services/cloudformation/provider.py

```diff
@@ -154,10 +157,36 @@ def __init__(self, message=None):
 
 
 class CloudformationProvider(CloudformationApi):
+    def _stack_status_is_active(self, stack_status: str) -> bool:
+        return stack_status not in [StackStatus.DELETE_COMPLETE]
+
     @handler("CreateStack", expand=False)
     def create_stack(self, context: RequestContext, request: CreateStackInput) -> CreateStackOutput:
         # TODO: test what happens when both TemplateUrl and Body are specified
         state = get_cloudformation_store(context.account_id, context.region)
+
+        stack_name = request.get("StackName")
+
+        # get stacks by name
+        active_stack_candidates = [
+            s
+            for s in state.stacks.values()
+            if s.stack_name == stack_name and self._stack_status_is_active(s.status)
+        ]
+
+        # TODO: fix/implement this code path
+        #   this needs more investigation how Cloudformation handles it (e.g. normal stack create or does it create a separate changeset?)
+        # REVIEW_IN_PROGRESS is another special status
+        # in this case existing changesets are set to obsolete and the stack is created
+        # review_stack_candidates = [s for s in stack_candidates if s.status == StackStatus.REVIEW_IN_PROGRESS]
+        # if review_stack_candidates:
+        # set changesets to obsolete
+        # for cs in review_stack_candidates[0].change_sets:
+        #     cs.execution_status = ExecutionStatus.OBSOLETE
+
```

Comment:
I think it's fine to leave this as a TODO for now. We already have much improved parity with this PR

Reply:
Yeah  I think it's better to first remove the inheritance and rework the change set entity a bit before tackling this.